server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectPolicyOperations.java

```diff
@@ -0,0 +1,348 @@
+    LOG.info(
+        "Received associate policies request for object type: {}, full name: {} under metalake: {}",
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            request.validate();
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+            String[] policyNames =
+                policyDispatcher.associatePoliciesForMetadataObject(
+                    metalake, object, request.getPoliciesToAdd(), request.getPoliciesToRemove());
+            policyNames = policyNames == null ? new String[0] : policyNames;
+
+            LOG.info(
+                "Associated policies: {} for object type: {}, full name: {} under metalake: {}",
+                Arrays.toString(policyNames),
+                type,
+                fullName,
+                metalake);
+            return Utils.ok(new NameListResponse(policyNames));
+          });
+
+    } catch (Exception e) {
+      return ExceptionHandlers.handlePolicyException(OperationType.ASSOCIATE, "", fullName, e);
+    }
+  }
+
+  private boolean hasConflictExclusivePolicy(
+      String metalake, MetadataObject object, Policy targetPolicy) {
+    if (!targetPolicy.exclusive()) {
+      return false;
+    }
+
+    Policy[] policies = policyDispatcher.listPolicyInfosForMetadataObject(metalake, object);
+    return Arrays.stream(policies)
+        .anyMatch(
+            p ->
+                p.exclusive()
+                    && p.policyType().equalsIgnoreCase(targetPolicy.policyType())
+                    && !p.name().equals(targetPolicy.name()));
+  }
+
+  private boolean hasConflictExclusivePolicy(Policy[] policies, Policy targetPolicy) {
+    if (!targetPolicy.exclusive() || ArrayUtils.isEmpty(policies)) {
+      return false;
+    }
+    return Arrays.stream(policies)
+        .anyMatch(
+            p ->
+                p.exclusive()
+                    && p.policyType().equalsIgnoreCase(targetPolicy.policyType())
+                    && !p.name().equals(targetPolicy.name()));
+  }
+
+  private Response logNotFoundPolicy(
+      String metalakeName, String policyName, String objectType, String fullName) {
+    LOG.warn(
+        "Policy {} not found for object type: {}, full name: {} under metalake: {}",
+        policyName,
+        objectType,
+        fullName,
+        metalakeName);
+    return Utils.notFound(
+        NoSuchPolicyException.class.getSimpleName(),
+        "Policy not found: "
+            + policyName
+            + " for object type: "
+            + objectType
+            + ", full name: "
+            + fullName
+            + " under metalake: "
+            + metalakeName);
+  }
+
+  private Optional<Policy> getPolicyForObject(
+      String metalake, MetadataObject object, String policyName) {
+    try {
+      return Optional.ofNullable(
```

Comment:
Since it will throw a `NoSuchPolicyException`, why do you use `Optional` here?

Reply:
Optional means that the policy exists but is not associated with the object.